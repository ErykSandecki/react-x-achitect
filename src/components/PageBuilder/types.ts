// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
