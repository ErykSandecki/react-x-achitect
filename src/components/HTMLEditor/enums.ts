export enum BlockHeight {
  small = 50,
  medium = 70,
  large = 100,
}

export enum BlockWidth {
  small = 50,
  medium = 200,
  large = 350,
}

export enum MouseMode {
  comment = 'comment',
  default = 'default',
  menu = 'menu',
  move = 'move',
}
