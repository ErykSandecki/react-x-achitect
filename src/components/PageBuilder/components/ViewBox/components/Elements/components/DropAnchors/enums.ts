export enum DropAnchorsPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
