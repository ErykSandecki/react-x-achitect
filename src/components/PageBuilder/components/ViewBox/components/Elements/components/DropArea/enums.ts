export enum DropAreaPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
