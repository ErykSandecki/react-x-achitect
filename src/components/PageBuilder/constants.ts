export const translationNameSpace = 'pageBuilder';

export const TOOLBAR_HEIGHT = 56;
