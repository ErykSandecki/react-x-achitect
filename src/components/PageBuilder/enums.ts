export enum Constrain {
  bottom = 'bottom',
  centerHorizontal = 'centerHorizontal',
  centerVertical = 'centerVertical',
  left = 'left',
  right = 'right',
  top = 'top',
}
