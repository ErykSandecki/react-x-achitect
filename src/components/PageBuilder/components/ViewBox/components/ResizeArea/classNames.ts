import { mapValues } from 'lodash';

// types
import { Anchor } from 'store/pageBuilder/enums';

export const className = 'ResizeArea';

export const classNames = {
  [className]: className,
  anchor: {
    name: `${className}__anchor`,
    modificators: {
      ...mapValues(Anchor, (anchor) => `${className}__anchor--${anchor}`),
    },
  },
};
