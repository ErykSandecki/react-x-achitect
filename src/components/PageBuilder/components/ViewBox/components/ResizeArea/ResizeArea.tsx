import { FC } from 'react';

// hooks
import { useResizeAreaEvents } from './hooks/useResizeAreaEvents';
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// types
import { Anchor } from 'store/pageBuilder/enums';
import { MouseMode } from 'components/PageBuilder/enums';
import { TElement } from 'types';

// styles
import styles from './resize-area.scss';

// utils
import { enumToArray } from 'utils';

export type TResizeAreaProps = {
  height: TElement['height'];
  id: TElement['id'];
  moseMode: MouseMode;
  width: TElement['width'];
  x: TElement['positionAbsolute']['x'];
  y: TElement['positionAbsolute']['y'];
};

const ResizeArea: FC<TResizeAreaProps> = ({
  height,
  id,
  moseMode,
  width,
  x,
  y,
}) => {
  const events = useResizeAreaEvents(height, id, moseMode, width, x, y);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <div className={cx(classNamesWithTheme[className])}>
      {enumToArray(Anchor).map((anchor) => (
        <div
          className={cx(
            classNamesWithTheme.anchor.name,
            classNamesWithTheme.anchor.modificators[
              anchor as keyof typeof Anchor
            ],
          )}
          key={anchor as keyof typeof Anchor}
          onMouseDown={(event) => events.onMouseDown(anchor as Anchor, event)}
        />
      ))}
    </div>
  );
};

export default ResizeArea;
