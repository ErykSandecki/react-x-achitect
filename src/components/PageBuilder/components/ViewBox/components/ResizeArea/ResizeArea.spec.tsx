import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import ResizeArea from './ResizeArea';

// mocks
import { selectedElementMock } from 'test/mocks/reducer/pageBuilderMock';
import 'test/mocks/sagas/allSagas';

// store
import { configureStore } from 'store/store';

// types
import { MouseMode } from 'components/PageBuilder/enums';

describe('Corners snapshots', () => {
  it('should render Corners', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ResizeArea
          height={100}
          id={selectedElementMock.id}
          moseMode={MouseMode.default}
          width={100}
          x={100}
          y={100}
        />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
