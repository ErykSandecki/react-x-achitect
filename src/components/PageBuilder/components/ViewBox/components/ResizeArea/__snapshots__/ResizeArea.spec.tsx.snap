// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Corners snapshots should render Corners 1`] = `
<DocumentFragment>
  <div
    class="ResizeArea ResizeArea--light"
  >
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--east ResizeArea__anchor--east--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--none ResizeArea__anchor--none--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--north ResizeArea__anchor--north--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--northEast ResizeArea__anchor--northEast--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--northWest ResizeArea__anchor--northWest--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--south ResizeArea__anchor--south--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--southEast ResizeArea__anchor--southEast--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--southWest ResizeArea__anchor--southWest--light"
    />
    <div
      class="ResizeArea__anchor ResizeArea__anchor--light ResizeArea__anchor--west ResizeArea__anchor--west--light"
    />
  </div>
</DocumentFragment>
`;
